packages/datadog-plugin-aws-sdk/src/helpers.js

```diff
@@ -0,0 +1,57 @@
+'use strict'
+
+const DEFAULT = 'default'
+const Services = {
+  cloudwatchlogs: require('./services/cloudwatchlogs'),
+  dynamodb: require('./services/dynamodb'),
+  kinesis: require('./services/kinesis'),
+  s3: require('./services/s3'),
+  redshift: require('./services/redshift'),
+  sns: require('./services/sns'),
+  sqs: require('./services/sqs'),
+  [DEFAULT]: require('./services/base')
+}
+
+function getService (serviceName) {
+  if (Services[serviceName]) {
+    return new Services[serviceName]()
+  } else {
+    return new Services[DEFAULT]()
```

Comment:
Why not just do the base stuff outside of the class hierarchy? Then all classes simply have to implement the right interface.

Reply:
It feels like there are two types of service tags(and values): default and 'extra'. I wanted to make a clean way to ensure the 'default' tags are always present but the 'extra' tags takes precedence if present. Doing this via a "base" service object, and then adding/overwriting that objects' keys/values depending on the "extra" service specific stuff, _before_ calling span.addTags felt much cleaner then, say, calling span.addTags multiple times, or having to check the span's tags in and then conditionally call span.addTags, or having to generate and pass in a "default tags" object to the service method?

Reply:
after syncing, updated to just do the base calss work within `helpers.addServiceTags` and removed the class hierarchy